src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift

```diff
@@ -141,93 +248,121 @@ class DiaryOverviewDayTableViewCell: UITableViewCell {
 
 			entryStackView.addArrangedSubview(imageView)
 			entryStackView.addArrangedSubview(entryLabelStackView)
+			encountersVisitsContainerStackView.addArrangedSubview(entryStackView)
 
-			encountersVisitsStackView.addArrangedSubview(entryStackView)
-		}
+			// Let's draw a seperator line between each entryStackView.
+			let separatorLine = UIView()
+			separatorLine.backgroundColor = .enaColor(for: .hairline)
+			separatorLine.translatesAutoresizingMaskIntoConstraints = false
+
+			// The position is dynamical and so we need to calculate the postion to our own (depending of the layout in the xib and what we do before this step)
+			let entryHeight = CGFloat(30) + imageView.frame.height
+			encountersVisitsContainerStackView.spacing = entryHeight
+
+			entryStackView.addSubview(separatorLine)
 
-		encountersVisitsContainerStackView.isHidden = encountersVisitsStackView.arrangedSubviews.isEmpty
+			// Draw the seperator line from leading to trailing of the encountersVisitsContainerStackView
+			NSLayoutConstraint.activate([
+				separatorLine.heightAnchor.constraint(equalToConstant: 1),
+				separatorLine.leadingAnchor.constraint(equalTo: encountersVisitsContainerStackView.leadingAnchor),
+				separatorLine.trailingAnchor.constraint(equalTo: encountersVisitsContainerStackView.trailingAnchor)
+			])
+
+			// Only for the first entryStackView we need the seperator line on the top of the container itself. Otherwise, we draw it on our own calculated position.
+			if index == 0 {
+				separatorLine.topAnchor.constraint(equalTo: encountersVisitsContainerStackView.topAnchor).isActive = true
+			} else {
+				separatorLine.topAnchor.constraint(equalTo: entryStackView.centerYAnchor, constant: -(entryHeight)).isActive = true
+			}
```

Comment:
you can store the constraint in a property an activate it inside of `NSLayoutConstraint.activate(` too

Reply:
Good idea 👍